### ISSUE TITLE
AVRO schema not supported if not started with {

### ISSUE THREAD

**Issue by chenww** [confluentinc/confluent-kafka-python#1198]
Description
===========

I have an AVRO schema which is union type. It failed to produce message to with SerializingProducer from AvroSerializer due to this code https://github.com/confluentinc/confluent-kafka-python/blob/23b4142a3b1f779ca31b6d082c37a51e622346e3/src/confluent_kafka/schema_registry/avro.py#L65

as above code will revise my schema and make it start with '{'.

How to reproduce
================
Design an avro schema like this:

`
[{
    "fields": [
      {
        "doc": "The name.",
        "name": "name",
        "type": "string"
      }]}]
 `

Checklist
=========
Please provide the following information:

 - [x] confluent-kafka-python and librdkafka version (`confluent_kafka.version()` and `confluent_kafka.libversion()`):
 confluent-kafka-python: 1.7.0

 - [ ] Apache Kafka broker version:
 - [ ] Client configuration: `{...}`
 - [ ] Operating system:
 - [x] Provide client logs (with `'debug': '..'` as necessary)

> 
ERROR    Expecting ',' delimiter: line 1 column 13 (char 12)
Traceback (most recent call last):
  ...
  File \"/opt/stackstorm/virtualenvs/modem_clone/lib/python3.7/site-packages/confluent_kafka/schema_registry/avro.py\", line 176, in __init__
    schema_dict = loads(schema.schema_str)
  File \"/usr/local/lib/python3.7/json/__init__.py\", line 348, in loads
    return _default_decoder.decode(s)
  File \"/usr/local/lib/python3.7/json/decoder.py\", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File \"/usr/local/lib/python3.7/json/decoder.py\", line 353, in raw_decode
    obj, end = self.scan_once(s, idx)
json.decoder.JSONDecodeError: Expecting ',' delimiter: line 1 column 13 (char 12)
>
 - [ ] Provide broker log excerpts
 - [ ] Critical issue

**Comment by chenww** [comment]
@OneCricketeer you are correct it needs more fields for completeness. The point here it shows the format matters. with '[' it will blow up

**Comment by chenww** [comment]
you are right. We are using Union type here which has the '[' representation as array. Let me update for cleaness

**Comment by joekrie** [comment]
I'm running into the same issue, and it's because the schema is a union. For example: 
```json
[
  "null",
  {
    "name": "person",
    "type": "record",
    "fields": [
      {
        "name": "first_name",
        "type": "string"
      },
      {
        "name": "last_name",
        "type": "string"
      }
    ]
  }
]
```
When the `_schema_loads` function in avro.py wraps it, the string it returns is no longer valid JSON, resulting in the error described by the OP.

**Comment by OneCricketeer** [comment]
IMO, there's no reason to define your schemas as A union of null and their type. The serializer should directly pass null values through without using any schema information

**Comment by joekrie** [comment]
I agree, but we're dealing with schemas that are auto-generated based on C# classes

**Comment by joekrie** [comment]
In my case I'm trying to deserialize messages in Python that were serialized and had their schemas generated in C#

**Comment by OneCricketeer** [comment]
AFAIK, you're welcome to define your own "reader schema". It's not required to use the one that's generated by a producer or one in the registry

**Comment by joekrie** [comment]
It looks like this was resolved by #1162

**Comment by chenww** [comment]
@joekrie thanks for the follow up
